### DISCUSSION TITLE
网站自动部署

### DISCUSSION THREAD

**Discussion by YDX-2147483647** [BITNP/BIThesis#409] [Ideas]
> [!NOTE]
>
> 转自 [doc: 补充学习资源 by YDX-2147483647 · Pull Request #210 · BITNP/BIThesis-wiki](https://github.com/BITNP/BIThesis-wiki/pull/210)。
>
> 现在更新不是特别频繁，所以我先建到 discussions 里了。

## 现在的方式

- BITNP/BIThesis-wiki 本身没有关联任何托管网站；
- fky2015/BIThesis-wiki 会自动定期拉取最新的上游代码，然后关联到 fky2015 的 Vercel 里面。

## 其它方案

- Vercel

  免费版不支持组织账号。

- Netlify

  > 之前曾经用过 Netlify，但是因为那个一旦超过配额就开始收费，我有点担心就不用了。（好几年前了）

- CloudFlare Pages

- GitHub Pages

  很难支持预览 PR。

## BIT101 的经验

> BIT101的方案可见：[BIT101结课文档-持续部署](https://bit101-project.feishu.cn/wiki/WHYpwtD4BiHGoskkyGEcruTInFd#QzaadMh9kopwxLxfy94cqU96nKJ)，当前主站用的是阿里云OSS，不过会产生一定流量费用。
>
> 个人最建议的方案是 CloudFlare Pages ，自动部署，访问速度也还不错，只是好像 PR 预览机器人的功能有点问题（不过应该问题也不算太大？）

现在 BIT101 是预览 PR 用 Netlify，主站用阿里云香港OSS，测试站用 CloudFlare Pages。

**Comment by fky2015** [comment]
那可以考虑切换到 CloudFlare Pages

**Comment by YDX-2147483647** [comment]
用 [Pull Request #263 · BITNP/BIThesis-wiki](https://github.com/BITNP/BIThesis-wiki/pull/263) 试了一下。

- 免费计划每月500次构建。

- [Preview deployments](https://developers.cloudflare.com/pages/configuration/preview-deployments/) 目前需要在主仓库有分支才行，不能只在fork有。

  > Commits/PRs from forked repositories will not create a preview. Support for this will come in the future.
  >
  > —[Known issues · Cloudflare Pages docs](https://developers.cloudflare.com/pages/platform/known-issues/#builds-and-deployment)

- https://dash.cloudflare.com 上可以添加成员。

![图片](https://github.com/BITNP/BIThesis/assets/73375426/cea18a7d-be77-4572-aa90-02eefd2a818e)
![图片](https://github.com/BITNP/BIThesis/assets/73375426/0c4ed27e-61fe-4e28-9970-5bb3723b1243)

Production: https://bithesis-wiki.pages.dev/

Commit preview: https://c89fed41.bithesis-wiki.pages.dev/

Branch preview: https://clone.bithesis-wiki.pages.dev/

**Reply by fky2015** [reply]
有些心动

**Comment by YDX-2147483647** [comment]
也看了一下 netlify，下次PR试试。

https://main--bithesis-wiki.netlify.app/

---

Cloudflare 也有评论，但只有从源仓库发出PR时才有，例如 https://github.com/BITNP/BIThesis-wiki/pull/267 。

而 Netlify 无此限制，例如 https://github.com/BITNP/BIThesis-wiki/pull/271 。

**Comment by YDX-2147483647** [comment]
[【讨论】是否需要从 Vercel 迁移 wiki 到 Cloudflares · Issue #320 · BITNP/BIThesis-wiki](https://github.com/BITNP/BIThesis-wiki/issues/320)

**Comment by YDX-2147483647** [comment]
https://github.com/BITNP/BIThesis-wiki/issues/320#issuecomment-2424021331
